File: frontend/src/types/PropertyListing.ts
import { PropertyTypeEnum } from "./enums/PropertyTypeEnum";

export interface PropertyListing {
  propertyId: number;
  title: string;
  price: number;
  bedrooms: number;
  bathrooms: number;
  sqFt: number;
  description?: string;
  address: string;
  listingDate: Date;
  propertyType: PropertyTypeEnum;
  imageUrl?: string;
}

File: Backend/Models/PropertyListing.cs
namespace Backend.Models
{
    public class PropertyListing
    {
        public int PropertyId { get; set; }
        public string Title { get; set; }
        public decimal Price { get; set; }
        public int Bedrooms { get; set; }
        public int Bathrooms { get; set; }
        public int SqFt { get; set; }
        public string? Description { get; set; }
        public string Address { get; set; }
        public DateTime ListingDate { get; set; }

        // property type (e.g., House, Condo, Apartment)
        public PropertyTypeEnum PropertyType { get; set; }
        public string? ImageUrl { get; set; }
    }
}